webpack/ForemanInventoryUpload/SubscriptionsPageExtension/SubscriptionsPageExtensionActions.js

```diff
@@ -0,0 +1,44 @@
+import API from 'foremanReact/API';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { foremanUrl } from '../../ForemanRhCloudHelpers';
+import { inventoryUrl } from '../ForemanInventoryHelpers';
+
+export const subscriptionCountListener = (items, unsubscribe) => {
+  if (items === undefined || items === 0) {
+    return;
+  }
+
+  fetchInventoryAutoUploadSetting(showToast, unsubscribe);
+};
+
+const showToast = (autoUploadEnabled, unsubscribe) => {
+  if (autoUploadEnabled === undefined || autoUploadEnabled) {
+    return;
+  }
+
+  window.tfm.toastNotifications.notify({
+    message: __('Now you can enable Red Hat inventory upload'),
+    type: 'info',
+    link: {
+      children: __('Go to inventory upload settings'),
+      href: foremanUrl('foreman_rh_cloud/inventory_upload'),
+    },
+  });
+
+  unsubscribe();
+};
+
+const fetchInventoryAutoUploadSetting = async (
+  processApiResponse,
+  unsubscribe
+) => {
```

Comment:
you can pass here one `callback` param which will be called in the `try` section.
it will let the consumer pass whatever they want.

example of usage:
```js
`fetchInventoryAutoUploadSetting(showToast)` 
//which already have `unsubscribe` if you init it inside `subscriptionCountListener`
```
this method can be changed a bit into:
```js
const fetchInventoryAutoUploadSetting = async callback => {
  try {
    const {
      data: { autoUploadEnabled },
    } = await API.get(inventoryUrl('auto_upload'));
    callback(autoUploadEnabled);
  } catch(error) {
    console.error(error)
  }
};
```

Reply:
I want to call the callback even if the API errored. This way I make sure unsubscribe is called, and I don't leave a dangling event listener.